File: 2. Silver/10610/정지훈_30.cs
namespace Algorithm
{
    internal class Program
    {
        static void Main(string[] args)
        {
            StreamReader sr = new(new BufferedStream(Console.OpenStandardInput()));
            StreamWriter sw = new(new BufferedStream(Console.OpenStandardOutput()));

            bool isTen = false;
            int Sum = 0;
            string Input = sr.ReadLine();

            List<char> listInputs = new List<char>();

            foreach (var chr in Input)
            {
                if ('0' == chr)
                    isTen = true;

                Sum += (chr - '0');
                listInputs.Add(chr);
            }

            if (isTen && (0 == Sum % 3))
            {
                listInputs.Sort();
                listInputs.Reverse();

                foreach (var chr in listInputs)
                    sw.Write(chr);
            }
            else
                sw.WriteLine(-1);
            
            sr.Close();
            sw.Close();
        }
    }
}

File: 2. Silver/10610/정지훈_30.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int Sum = 0;
	bool isTen = false;
	string Input;
	cin >> Input;

	sort(Input.begin(), Input.end(), greater<char>());

	for (int i = 0; i < Input.length(); ++i)
	{
		if ('0' == Input[i])
			isTen = true;

		Sum += static_cast<int>(Input[i]);
	}

	if (isTen && !(Sum % 3))
		cout << Input << '\n';
	else
		cout << -1 << '\n';

	return 0;
}